### PR TITLE
Ensure Kotlin dependencies are properly handled when present in a configuration with no parents

### PR DESCRIPTION
**Pull request by tresat**
This will handle the case of Kotlin plugins (applied to the parentless classpath configuration).

### CODE REVIEW

**Comment by ben-manes** [comment]
odd, the test suite breaks with this change

**Comment by tresat** [comment]
You're right, this breaks additional tests.

I was getting failures in `DifferentGradleVersionsSpec` when I checked out `master` and ran `build` for the first time, so I assumed they weren't related.  Then I didn't run the full test suite after that, since it would always fail.

I'll take another look.

**Comment by ben-manes** [comment]
iirc, that happens when you run with a newer jdk. You can use jenv, asdf, etc. to set a local one for the project. Trying right now with jdk11 on master to see if it passes locally. I don't recall what the issue was, but remember being confused by that test failure too

**Comment by ben-manes** [comment]
yep, jdk11 worked locally

File: gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt [hunk 1]
```diff
@@ -89,11 +89,12 @@ class Resolver(
...
   ): Configuration {
-    // Kotlin deps anywhere in the hierarchy are a special case we'll handle later
-    val kotlinDeps = { dependency: ExternalDependency -> dependency.group == "org.jetbrains.kotlin" && dependency.version != null }
+    // Kotlin deps anywhere in the hierarchy are a special case we'll handle later separately, unless this configuration
+    // has no parents (such as the classpath conf, which contains plugins, including kotlin ones), in which case handle those deps here
+    val kotlinDeps = { dependency: ExternalDependency -> (dependency.group?.startsWith("org.jetbrains.kotlin") ?: false) && dependency.version != null }
     val latest = configuration.allDependencies
       .filterIsInstance<ExternalDependency>()
-      .filterNot(kotlinDeps)
+      .filter { configuration.hierarchy.size > 1 && kotlinDeps(it) }
```
Comment (ben-manes) [inline gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt:97]: maybe `filterNot`? trying locally

Comment (ben-manes) [reply gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt:97]: passes except for your new test, which I think might be a formatting difference?

```console
PluginUpdateDetectionSpec > kotlin plugin in the implementation configuration is properly handled FAILED
    Condition not satisfied:

    result.output.contains """The following dependencies have later milestone versions: - org.jetbrains.kotlin:kotlin-gradle-plugin [$KOTLIN_VERSION -> """
    |      |      |                                                                                                                   |
    |      |      false                                                                                                               1.6.0
    |
    |      > Task :dependencyUpdates
    |
    |      ------------------------------------------------------------
    |      : Project Dependency Updates (report to plain text file)
    |      ------------------------------------------------------------
    |
    |      The following dependencies are using the latest milestone version:
    |       - org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.0
    |       - org.jetbrains.kotlin:kotlin-stdlib:1.6.0
```

Comment (tresat) [reply gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt]: I think detecting by `isForced` might be what's needed.

**Comment by tresat** [comment]
A little more context: I saw the comments about plugins trying to force versions, so that led me to this path.  My approach to fixing this has been to preserve the special case logic for `inheritedKotlin` by making the smallest change that fixes these new test cases and leaving that alone.  

But after talking with @jvandort, it seems `isForce` isn't doing anything in newer versions of Gradle, and there is no way remaining for external deps to be marked `forced`.  So the effect of this change is to just always allow Kotlin deps to be included in that first loop that collects them into `latest`.

So I'm wondering if _any_ of the special casing for Kotlin is still necessary?  If I remove it all, I get the same passing build.  Perhaps it is no longer useful?  I couldn't find any existing tests that hinted at why it was still there.  Do you know how we could create one to confirm if it is needed?

**Comment by ben-manes** [comment]
Here is the commit that introduced it, https://github.com/ben-manes/gradle-versions-plugin/commit/0a290fb86d8ba3962fead9d3c2be8fa91f1ab5e5, which seems to be trying to fix #424.

If we don't need it anymore that is better. wdyt?

**Comment by ben-manes** [comment]
I released to unblock, but happy to revisit the removal of that code if its useless nowadays.

**Comment by tresat** [comment]
@ben-manes Can you take a look at https://github.com/ben-manes/gradle-versions-plugin/pull/826 when you get a chance?  I think this is all that is needed.  I added tests for the kotlin stdlib dependency that was the origin of the special case in the `Resolver`.

**Pull request merged**